src/anomalib/models/components/base/anomaly_module.py

```diff
@@ -259,6 +199,29 @@ def configure_transforms(self, image_size: tuple[int, int] | None = None) -> Tra
             ],
         )
 
+    @property
+    def post_processor(self) -> PostProcessor | None:
+        """Return the post processor."""
+        return self._post_processor
+
+    def set_post_processor(self, post_processor: PostProcessor) -> None:
+        """Set the post processor.
+
+        We cannot use a setter here, because the post processor is an nn.Module.
+        """
+        self._post_processor = post_processor
+
+    def default_post_processor(self) -> PostProcessor:
+        """Default post processor.
+
+        Override in subclass for model-specific post-processing behaviour.
+        """
+        if self.learning_type == LearningType.ONE_CLASS:
+            return OneClassPostProcessor()
+        msg = f"No default post-processor available for model {self.__name__} with learning type {self.learning_type}. \
+              Please override the default_post_processor method in the model implementation."
+        raise NotImplementedError(msg)
+
```

Comment:
We can't use a setter because `PostProcessor` is a subclass of `nn.Module` which can't be assigned with a setter. The additional `default_post_processor` is needed because the engine is responsible for setting the post_processor in the model. If the getter would return the default post_processor, the engine would not be able to determine if the retrieved post processor has been explicitly set in a previous run or if it's just the default post-processor for the model. This behavious is similar to that of transforms, where we also have the 3 methods.